calibr8/core.py

```diff
@@ -727,71 +826,13 @@ def infer_independent(
-            func=lambda x: self.likelihood(x=x, y=y),
-            # by restricting the integral into the interval [a,b], the resulting PDF is
-            # identical to the posterior with a Uniform(a, b) prior.
-            # 1. prior probability is constant in [a,b]
-            # 2. prior probability is 0 outside of [a,b]
-            # > numerical integral is only computed in [a,b], but because of 1. and 2., it's
-            #   identical to the integral over [-∞,+∞]
-            a=lower,
-            b=upper,
-        )
-
-        # high resolution x-coordinates for integration
-        # the first integration is just to find the peak
-        x_integrate = numpy.linspace(lower, upper, 10_000)
-        area = scipy.integrate.cumtrapz(
-            self.likelihood(x=x_integrate, y=y, scan_x=True), x_integrate, initial=0
-        )
-        cdf = area / area[-1]
-
-        # now we find a high-resolution CDF for (1-shrink) of the probability mass
-        shrink = 0.00001
-        xfrom, xto = _get_eti(x_integrate, cdf, 1 - shrink)
-        x_integrate = numpy.linspace(xfrom, xto, 100_000)
-        area = scipy.integrate.cumtrapz(
-            self.likelihood(x=x_integrate, y=y, scan_x=True), x_integrate, initial=0
-        )
-        cdf = (area / area[-1]) * (1 - shrink) + shrink / 2
-
-        # TODO: create a smart x-vector from the CDF with varying stepsize
-
-        if ci_prob != 1:
-            if not isinstance(ci_prob, (int, float)) or not (0 < ci_prob <= 1):
-                raise ValueError(
-                    f"Unexpected `ci_prob` value of {ci_prob}. Expected float in interval (0, 1]."
-                )
-
-            # determine the interval bounds from the high-resolution CDF
-            eti_lower, eti_upper = _get_eti(x_integrate, cdf, ci_prob)
-            hdi_lower, hdi_upper = _get_hdi(x_integrate, cdf, ci_prob, eti_lower, eti_upper, history=None)
-
-            eti_x = numpy.linspace(eti_lower, eti_upper, steps)
-            hdi_x = numpy.linspace(hdi_lower, hdi_upper, steps)
-            eti_pdf = self.likelihood(x=eti_x, y=y, scan_x=True) / likelihood_integral
-            hdi_pdf = self.likelihood(x=hdi_x, y=y, scan_x=True) / likelihood_integral
-            eti_prob = _interval_prob(x_integrate, cdf, eti_lower, eti_upper)
-            hdi_prob = _interval_prob(x_integrate, cdf, hdi_lower, hdi_upper)
-        else:
-            x = numpy.linspace(lower, upper, steps)
-            eti_x = hdi_x = x
-            eti_pdf = hdi_pdf = self.likelihood(x=x, y=y, scan_x=True) / likelihood_integral
-            eti_prob = hdi_prob = 1
-
-        median = x_integrate[numpy.argmin(numpy.abs(cdf - 0.5))]
-
-        return ContinuousUnivariateInference(
-            median,
-            eti_x=eti_x,
-            eti_pdf=eti_pdf,
-            eti_prob=eti_prob,
-            hdi_x=hdi_x,
-            hdi_pdf=hdi_pdf,
-            hdi_prob=hdi_prob,
+        return infer_independent(
+            likelihood=self.likelihood,
```

Comment:
although it makes no difference, I would prefer the order as specified above (although you use kwargs only)

Reply:
I don't understand, this is the same order as in both the method and the function signatures?

Reply:
no, y comes before likelihood in the function above? 
y is positional and the rest kw-based, but here you insert likelihood before y. 

No change in the end, I just don't find it very pretty :)